### PR TITLE
Fix player's movement

### LINKED ISSUE #457: [BUG] We are lossing status from the back
This is what causes the characters to walk funny (they appear to be limping)

### PR DESCRIPTION
**Pull request by tvillegas98**
closes #457

In this PR, we're changing the way we turn on/off the walking animation. Formerly we we're using an if that changes its states regarding in position differences and sometimes that caused to have a choppy movement for every character. Now we're going to turn on the animation if the users is pressing any kind of input (either visual or physical). For other players we still using the former logic.

Disclaimer:
This PR fixes the neon crashing that was not working while the piercing effect is on

### CODE REVIEW

File: client/Assets/Scripts/PlayerMovement.cs [hunk 1]
```diff
@@ -484,10 +505,8 @@ private void movePlayer(GameObject player, Player playerUpdate)
                 {
                     characterOrientation.ForcedRotationDirection = movementDirection;
                 }
-
-                // TODO: why not use character state?
-                walking = true;
             }
+            walking = true;
```
Comment (manucamejo) [inline client/Assets/Scripts/PlayerMovement.cs:509]: Should we remove this?

File: server/native/gamestate/src/game.rs [hunk 2]
```diff
@@ -219,6 +219,8 @@ impl GameState {
             return Ok(());
         }
 
+        player.action = PlayerAction::MOVING.clone();
```
Comment (manucamejo) [inline server/native/gamestate/src/game.rs:222]: Cloning is not necessary

File: server/native/gamestate/src/player.rs [hunk 3]
```diff
@@ -83,7 +83,7 @@ pub enum Status {
     DISCONNECTED,
 }
 
-#[derive(Debug, Clone, NifUnitEnum)]
+#[derive(rustler::NifTaggedEnum, Debug, Hash, Clone, PartialEq, Eq)]
```
Comment (manucamejo) [inline server/native/gamestate/src/player.rs:86]: If we remove the clone function, maybe we don't need these changes.

File: client/Assets/Animations/Muflus/MuflusController.controller [hunk 4]
```diff
@@ -619,7 +619,7 @@ AnimatorStateTransition:
   m_Name: 
   m_Conditions:
   - m_ConditionMode: 1
-    m_ConditionEvent: Skill3
+    m_ConditionEvent: Skill1
```
Comment (manucamejo) [inline client/Assets/Animations/Muflus/MuflusController.controller:622]: Is this necessary?

File: client/Assets/Scripts/EventsBuffer.cs [hunk 5]
```diff
@@ -42,4 +42,31 @@ public GameEvent getNextEventToRender(long pastTime)
             return nextGameEvent;
         }
     }
+
+    public bool playerIsMoving(ulong playerId, long pastTime)
+    {
+        var count = 0;
+        GameEvent previousRenderedEvent = this.getNextEventToRender(pastTime - 30);
+        GameEvent currentEventToRender = this.getNextEventToRender(pastTime);
+        GameEvent followingEventToRender = this.getNextEventToRender(pastTime + 30);
```
Comment (manucamejo) [inline client/Assets/Scripts/EventsBuffer.cs:51]: What happen if we don't have `following`?

File: client/Assets/Scripts/Messages.pb.cs [hunk 6]
```diff
@@ -698,7 +700,7 @@ public sealed partial class GameEvent : pb::IMessage<GameEvent>
       }
       WinnerPlayer.MergeFrom(other.WinnerPlayer);
     }
-    selectedCharacters_.MergeFrom(other.selectedCharacters_);
+    selectedCharacters_.Add(other.selectedCharacters_);
```
Comment (manucamejo) [inline client/Assets/Scripts/Messages.pb.cs:703]: Why do we need this?

File: client/Assets/Scripts/EventsBuffer.cs [hunk 7]
```diff
@@ -42,4 +42,31 @@ public GameEvent getNextEventToRender(long pastTime)
             return nextGameEvent;
         }
     }
+
+    public bool playerIsMoving(ulong playerId, long pastTime)
```
Comment (manucamejo) [inline client/Assets/Scripts/EventsBuffer.cs:46]: Please, document a little this function.

**Review by manucamejo — CHANGES_REQUESTED** [review]

File: client/Assets/Scripts/Messages.pb.cs (hunk 6 above)
Comment (tvillegas98) [reply client/Assets/Scripts/Messages.pb.cs:703]: Protobuf generated this not me

File: client/Assets/Scripts/EventsBuffer.cs (hunk 5 above)
Comment (tvillegas98) [reply client/Assets/Scripts/EventsBuffer.cs:51]: We will always have a following event, check the `getNextEventToRender`

File: client/Assets/Animations/Muflus/MuflusController.controller (hunk 4 above)
Comment (tvillegas98) [reply client/Assets/Animations/Muflus/MuflusController.controller:622]: @saugon Did this, fixed a bug where muflus couldn't do a barrel roll while walking

File: client/Assets/Scripts/PlayerMovement.cs (hunk 1 above)
Comment (tvillegas98) [reply client/Assets/Scripts/PlayerMovement.cs:509]: Yes, now we can remove it

File: server/native/gamestate/src/game.rs (hunk 2 above)
Comment (tvillegas98) [reply server/native/gamestate/src/game.rs]: Done!

File: server/native/gamestate/src/player.rs (hunk 3 above)
Comment (tvillegas98) [reply server/native/gamestate/src/player.rs]: Done!

File: client/Assets/Scripts/EventsBuffer.cs (hunk 7 above)
Comment (tvillegas98) [reply client/Assets/Scripts/EventsBuffer.cs:55]: D0one!

**Review by manucamejo — APPROVED** [review]

**Review by ncontinanza — CHANGES_REQUESTED** [review]
The movement is working fine, but the fixes from the commit `cc4f04200202cd9603f515fe36357ec5a35d09b4` seem to have lost their effect, and Muflus does not attack while walking, and H4ck does not stop moving to perform an attack.

**Review by manucamejo — APPROVED** [review]

**Review by ncontinanza — APPROVED** [review]

**Pull request merged**